test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/ResponseTests.cs

```diff
@@ -159,7 +159,7 @@ public class ResponseTests
                     {
                         await context.Response.WriteAsync("hello, world");
                         await context.Response.Body.FlushAsync();
-                        ex = Assert.Throws<InvalidOperationException>(() => context.Response.OnStarting(_ => TaskUtilities.CompletedTask, null));
+                        ex = Assert.Throws<BadHttpResponseException>(() => context.Response.OnStarting(_ => TaskUtilities.CompletedTask, null));
```

Comment:
I agree with @halter73 that this should continue throwing `InvalidOperationException`.